From: =?utf-8?q?Ond=C5=99ej_Sur=C3=BD?= <ondrej@sury.org>
Date: Fri, 24 Jun 2016 08:21:28 +0200
Subject: Adjust tidy extension for tidy-html5

---
 ext/tidy/tidy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/tidy/tidy.c b/ext/tidy/tidy.c
index 4e307eec0..8cecf1795 100644
--- a/ext/tidy/tidy.c
+++ b/ext/tidy/tidy.c
@@ -31,7 +31,7 @@
 #include "ext/standard/info.h"

 #include "tidy.h"
-#include "buffio.h"
+#include "tidybuffio.h"

 /* compatibility with older versions of libtidy */
 #ifndef TIDY_CALL